instructure/canvas-ios
# Fix split view bugs

refs: MBl-13352
affects: student
release note: Fixed layout issues with split view on iPad

* Trying to avoid `targetDisplayModeForAction`. It doesn't seem like we need to implement it and it's called 8 times on rotation on iOS 13. Who wants all that noise.
* Consolidating to one UISplitViewController subclass so that we don't have to maintain two
* Fixed some cases where displayModeButtonItem was not showing up when it should
* Call super in a few `viewWillTransition(to:coordinator:)` overrides because docs say to

**Test plan:**
* Tap around with iPad, expanding and collapsing split view
* Expand/collapse button should persist across navigation
* Enable and disable side-by-side and tap around the student app on iPad
